### PR TITLE
Handle empty dependency section in add_dependency()

### PR DESCRIPTION
**Pull request by msentissi**
### Description

- A `get_value()` call may return `None` when a default value is supplied, if the path exists and is simply an empty key.
This case was encountered when attempting to call `add_dependency()` to add a dependency to an empty (host or run) section.
- `_is_valid_dependency_path()` was not inclusive of test dependency paths.

### Solution

- Handle the edge-case before proceeding.
- Amend `_is_valid_dependency_path()` logic.

### CODE REVIEW

File: conda_recipe_manager/parser/recipe_parser_deps.py [hunk 1]
```diff
@@ -169,9 +169,10 @@ def add_dependency(
         patch_path = RecipeParserDeps._init_patch_path(dep, dep_mode, base_path, is_new_section)
 
         # TODO: Add a "get dependencies at path" function to `RecipeReaderDeps`
-        cur_deps: Final[list[Optional[str]]] = cast(
-            list[Optional[str]], self.get_value(base_path, sub_vars=True, default=[])
-        )
+        opt_cur_deps = self.get_value(base_path, sub_vars=True, default=[])
+        # Handle empty section case
+        opt_cur_deps = [] if opt_cur_deps is None else opt_cur_deps
+        cur_deps: Final[list[Optional[str]]] = cast(list[Optional[str]], opt_cur_deps)
```
[the remark this answers is not in the document]
Comment (schuylermartin45) [reply conda_recipe_manager/parser/recipe_parser_deps.py:175]: Namely it should have thrown an error on `enumerate(None)` below.

**Review by schuylermartin45 — COMMENTED** [review]
On top of the other comments made, we should add a regression test for this case and link the issue and/or PR number.

File: conda_recipe_manager/parser/recipe_parser_deps.py [hunk 2]
```diff
@@ -49,7 +49,7 @@ def _is_valid_dependency_path(dep_path: str) -> bool:
         if len_path == _SINGLE_OUTPUT_LEN:
             return (
                 not bool(path[0])
-                and path[1] == "requirements"
+                and path[1] in {"requirements", "test"}
```
[the remark this answers is not in the document]
Comment (msentissi) [reply conda_recipe_manager/parser/recipe_parser_deps.py]: Done.

**Comment by schuylermartin45** [comment]
I think I'm good with this, pending adding the regression test.

**Review by schuylermartin45 — APPROVED** [review]
